packages/story-editor/src/components/library/panes/shopping/shoppingPane.js

```diff
@@ -46,48 +51,160 @@ const HelperText = styled(Text).attrs({
+        height: 25,
+        product,
+      });
+
+      const PRODUCT_ADDED_TEXT = sprintf(
+        /* translators: %s: product title added to page.*/
+        __('%s added to page', 'web-stories'),
+        product.productTitle
+      );
+
+      speak(PRODUCT_ADDED_TEXT);
+    },
+    [insertElement, speak]
+  );
+
+  const deleteProduct = useCallback(
+    (product) => {
+      const element = currentPageProducts.find(
+        (item) => item.productId === product.productId
+      );
+      if (element) {
+        deleteElementById({ elementId: element.elementId });
+      }
+    },
+    [deleteElementById, currentPageProducts]
+  );
+
+  const onClick = useCallback(
+    (product, onPage) =>
+      onPage ? deleteProduct(product) : insertProduct(product),
+    [deleteProduct, insertProduct]
+  );
+
+  const handleClearInput = useCallback(() => {
+    setSearchTerm('');
+    getProductsByQuery();
+  }, [setSearchTerm, getProductsByQuery]);
 
-  if (!isEnabled) {
+  if (!isShoppingIntegrationEnabled) {
     return null;
   }
 
   return (
     <Pane id={paneId} {...props}>
-      <Row>
-        <HelperText>
-          {__(
-            'Select a product to add to the page. It will be displayed as a "dot" on the page, as well as in the page attachment.',
-            'web-stories'
-          )}
-        </HelperText>
-      </Row>
-      <Row>
-        <ProductDropdown product={product} setProduct={setProduct} />
-      </Row>
-      <Button
-        variant={BUTTON_VARIANTS.RECTANGLE}
-        type={BUTTON_TYPES.SECONDARY}
-        size={BUTTON_SIZES.SMALL}
-        onClick={insertProduct}
-        disabled={!product}
+      <Section
+        data-testid="products-library-pane"
+        title={__('Products', 'web-stories')}
       >
-        {__('Insert product', 'web-stories')}
-      </Button>
+        <Row>
+          <HelperText>
+            {__(
+              'This will add products as a tappable dot on your story.',
+              'web-stories'
+            )}
+          </HelperText>
+        </Row>
+        <Row>
+          <SearchInput
+            aria-label={__('Product search', 'web-stories')}
+            inputValue={searchTerm}
+            onChange={onSearch}
+            placeholder={__('Search', 'web-stories')}
+            onKeyDown={handleInputKeyPress}
+            onFocus={handleFocus}
+            isOpen
+            ariaClearLabel={__('Clear product search', 'web-stories')}
+            clearId="clear-product-search"
+            handleClearInput={handleClearInput}
+          />
+        </Row>
+        {isLoading || products?.length > 0 ? (
+          <ProductList
+            isLoading={isLoading}
+            isMenuFocused={isMenuFocused}
+            onClick={onClick}
+            products={products}
+            onPageProducts={currentPageProducts}
+          />
+        ) : searchTerm?.length ? (
+          <HelperText>{__('No products found.', 'web-stories')}</HelperText>
+        ) : null}
```

Comment:
```suggestion
        {isLoading ? (
              <Loading>
                  <Spinner>
                   <CircularProgress size={24} />
               </Spinner>
              </Loading>
          )}
          {!isLoading && products?.length > 0 ? (
            <ProductList
                isMenuFocused={isMenuFocused}
                onClick={onClick}
                products={products}
                onPageProducts={currentPageProducts}
               />
         )};
         {!isLoading && products?.length === 0 & searchTerm?.length ? (
          <HelperText>{__('No products found.', 'web-stories')}</HelperText>
        )}
```

This is a little cleaner. Current block is a little confusing and nested.